flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/FlinkSqlOperatorTable.java

```diff
@@ -224,6 +224,15 @@ public void lookupOperatorOverloads(
                     OperandTypes.or(OperandTypes.NUMERIC_INTEGER, OperandTypes.NUMERIC),
                     SqlFunctionCategory.NUMERIC);
 
+    public static final SqlFunction TRUNCATE =
+            new SqlFunction(
+                    "TRUNCATE",
+                    SqlKind.OTHER_FUNCTION,
+                    FlinkReturnTypes.ROUND_FUNCTION_NULLABLE,
```

Comment:
If we're reusing this strategy, the name doesn't seem appropriate anymore. We should rename this to something that explains what it's doing.

Reply:
If you look into this method you'll find that it is calling `LogicalTypeMerging#findRoundDecimalType`. In that method there is a line stating that
```java
// NOTE: rounding may increase the digits by 1, therefore we need +1 on precisions.
return new DecimalType(false, 1 + precision - scale + round, round);
```

However for `truncate` function the number of digits will not increase, thus `FlinkReturnTypes.ROUND_FUNCTION_NULLABLE` is not the best choice to use here.

What I'll suggest is that you create a new method in `LogicalTypeMerging` called `findTruncateDecimalType`. This method and `findRoundDecimalType` can together reuse some code. Then you might want to create `FlinkReturnTypes.TRUNCATE_FUNCTION_NULLABLE` which will also reuse a lot of code with `ROUND_FUNCTION_NULLABLE`.

Reply:
According to the second review request below, if round and truncate could use the same logic, could I refactor those method name to things like 'findRoundOrTruncateDecimalType'?

Reply:
I would prefer "ROUNDING_NULLABLE", as "rounding" is a more general behavior, not specified to a function.

Reply:
OK. I'll keep it the same as the round method.